uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/WorkPattern.php

```diff
@@ -45,6 +45,10 @@ public static function create($params) {
       $params['weight'] = self::getMaxWeight() + 1;
     }
 
+    if(!empty($params['id']) && isset($params['is_active']) && !$params['is_active']) {
```

Comment:
you called :

`empty($params['id'])`

multiple times in create method , you can group all of them is a single if statement like this :

```php
  $hook = 'edit';
    if(empty($params['id'])) {
    $hook = 'create';
      $params['weight'] = self::getMaxWeight() + 1;
 if (isset($params['is_active']) && !$params['is_active'])  {
self::checkIfPatternCanBeDisabled($params['id']);
}
    }
```